athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsMetadataHandler.java

```diff
@@ -0,0 +1,320 @@
+     */
+    @Override
+    public ListTablesResponse doListTables(BlockAllocator allocator, final ListTablesRequest request) throws Exception
+    {
+        Set<TableName> tables = new LinkedHashSet<>();
+        if (glueClient != null) {
+            try {
+                tables.addAll(super.doListTables(allocator,
+                        new ListTablesRequest(request.getIdentity(), request.getQueryId(), request.getCatalogName(),
+                                request.getSchemaName(), null, UNLIMITED_PAGE_SIZE_VALUE),
+                        TABLE_FILTER).getTables());
+            }
+            catch (RuntimeException e) {
+                LOGGER.warn("doListTables: Unable to retrieve tables from AWSGlue in database/schema {}", request.getSchemaName(), e);
+            }
+        }
+        return new ListTablesResponse(request.getCatalogName(), new ArrayList<>(tables), null);
+    }
+
+    /**
+     * Used to get definition (field names, types, descriptions, etc...) of a Table.
+     *
+     * @param blockAllocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request        Provides details on who made the request and which Athena catalog, database, and table they are querying.
+     * @return A GetTableResponse which primarily contains:
+     * 1. An Apache Arrow Schema object describing the table's columns, types, and descriptions.
+     * 2. A Set<String> of partition column names (or empty if the table isn't partitioned).
+     * 3. A TableName object confirming the schema and table name the response is for.
+     * 4. A catalog name corresponding the Athena catalog that was queried.
+     */
+    @Override
+    public GetTableResponse doGetTable(BlockAllocator blockAllocator, GetTableRequest request) throws Exception
+    {
+        GetTableResponse response;
+        if (glueClient != null) {
+            try {
+                response = super.doGetTable(blockAllocator, request);
+                if (null == response || null == response.getSchema() || response.getSchema().getFields().isEmpty() || (response.getSchema().getFields().stream().count() - response.getPartitionColumns().stream().count()) == 0) {
+                    throw new GcsConnectorException("doGetTable: Unable to retrieve schema from AWSGlue in database/schema. " + request.getTableName().getTableName());
+                }
+                return response;
+            }
+            catch (RuntimeException e) {
+                LOGGER.warn("doGetTable: Unable to retrieve table {} from AWSGlue in database/schema {}. " +
+                                "Falling back to schema inference. If inferred schema is incorrect, create " +
+                                "a matching table in Glue to define schema (see README)",
+                        request.getTableName().getTableName(), request.getTableName().getSchemaName(), e);
+            }
+        }
+        Table table = GlueUtil.getGlueTable(request, request.getTableName(), glueClient);
```

Comment:
if line 192 glueClient is null and we call `GlueUtil.getGlueTable` which underlying we called this
GetTableResult result = awsGlue.getTable(getTableRequest); 
We would result in null pointer? 

In addition, super.doGetTable and GlueUtil.getGlueTable does pretty much the same thing beside super.doGetTable has filter. why do we need GlueUtil.getGlueTable?